### DISCUSSION TITLE
Why does `socket.ack` always send back an array?

### DISCUSSION THREAD

**Discussion by JasterV** [Totodore/socketioxide#271] [Q&A] [answered]
Hi! I realized that when I send back an `ack` from the server it is always in the form of an array. Is that an expected behavior? I can't find it on the official Socket.io protocol spec.

In my case I just want the server to acknowledge back a single structured piece of data but the client gets an array with that data inside instead.

I would appreciate some help :D

**Comment by JasterV** [comment]
I've noticed that you actually enforce the transformation of an Ack packet to an array on the code

https://github.com/Totodore/socketioxide/blob/c8bbd2238a3e16b12cb1bd9af569b5a585acff82/socketioxide/src/packet.rs#L331

**Reply by Totodore** [reply, marked as answer]
It might be a miscomprehension, but according to the protocol an ack packet is always encoded as an array:
https://socket.io/docs/v4/socket-io-protocol/#acknowledgement-1

However it should be de-encoded as a single element in the frontend if you provided only one element. 
It would be worth to check the encoding code for ack packet in the official implementation.
